code/game/turfs/simulated/floor.dm

```diff
@@ -97,6 +102,11 @@
 /turf/open/floor/proc/update_icon()
 	update_visuals()
 	return 1
+	if(!update_visuals())
```

Comment:
@UndeadNappist

Reply:
update_visuals() is a proc, not a variable, so if(!update_visuals()) checks if it the proc returned false

Its the LINDA proc to update gas overlays, etc. https://github.com/BadDeathclaw/TG-Claw/blob/d6ffeb207849abcc3d993d4326ee0cef6c9c1a73/code/modules/atmospherics/environmental/LINDA_turf_tile.dm#L97


So you:
run update_icon() super procedure (previously it would return if the super returns false)
run update_visuals, return if it returns false (why?)
do broken/burnt/icon_state checks
run update_visuals again

Are you sure you didn't mean to do if(!..()) like the original in plasteel?
![image](https://user-images.githubusercontent.com/22902112/49628206-9761aa80-fa2e-11e8-8391-30fecbd43108.png)

Reply:
For whatever reason I assumed you could use procs interchangeably as variables, and that if I didn't include ..() it would ignore the previous inherited traits from update_icon higher up in the hierarchy which is used in all sorts of stuff and, hopefully eventually, could be refactored all the way up to /atom/ and minimize lines of code.

Reply:
Finally think I got it to an acceptable point.

Reply:
yes procs can be var if it returns either 0 or anything above 0